[std] Remove problematic phrases from notes (#6758)
cplusplus/draft: source/numerics.tex

if \tcode{X} cannot be converted to \tcode{int}
by integral promotion\iref{conv.prom}, the program is ill-formed.
\begin{note}

> Arguments that can be promoted to \tcode{int} are supported for compatibility with C.

Reviewer: Alternatively (but I don't think it's necessarily better):

```suggestion
Arguments that can be promoted to \tcode{int} provide compatibility with C.
```

Reviewer: Or "allowing arguments that can be promoted to `int` provide compatibility with C".

Author: Yes to the latter (it's not the arguments that provide compatibility, but this feature as a whole).
